gcloud/src/main/java/org/jimmutable/gcloud/search/IndexManager.java

```diff
@@ -0,0 +1,167 @@
+package org.jimmutable.gcloud.search;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.jimmutable.gcloud.logging.LogSupplier;
+
+import com.google.appengine.api.search.Document;
+import com.google.appengine.api.search.Field;
+import com.google.appengine.api.search.Index;
+import com.google.appengine.api.search.IndexSpec;
+import com.google.appengine.api.search.OperationResult;
+import com.google.appengine.api.search.PutException;
+import com.google.appengine.api.search.PutResponse;
+import com.google.appengine.api.search.Results;
+import com.google.appengine.api.search.ScoredDocument;
+import com.google.appengine.api.search.SearchException;
+import com.google.appengine.api.search.SearchServiceFactory;
+import com.google.appengine.api.search.StatusCode;
+
+/**
+ * Use this class to upsert, delete and get a document from a search Index
+ * 
+ * @author trevorbox
+ *
+ */
+public class IndexManager {
+
+	private static final Logger logger = Logger.getLogger(IndexManager.class.getName());
+
+	private static Index getIndex(IndexId indexId) {
+		IndexSpec indexSpec = IndexSpec.newBuilder().setName(indexId.getSimpleValue()).build();
+		return SearchServiceFactory.getSearchService().getIndex(indexSpec);
+	}
+
+	/**
+	 * Performs a get operation on the index for the given DocumentId
+	 * 
+	 * @param index_id
+	 *            IndexId
+	 * @param document_id
+	 *            DocumentId
+	 * @param default_value
+	 *            The Document to return in the event the document is not found
+	 * @return
+	 */
+	public static Document getComplex(IndexId index_id, DocumentId document_id, Document default_value) {
+		Index index = getIndex(index_id);
+		Document d = index.get(document_id.getSimpleValue());
+		if (d != null) {
+			return d;
+		}
+		return default_value;
+	}
+
```

Comment:
wrap all code in getComplex in a try/catch block, the code can through a RuntimeException (because networking).  The contract for getComplex requires you trap this and return default_value

Validate input params are not null (of course, default_value can be null)